File: src/main/java/ProbYDanioElevado.java
public class ProbYDanioElevado extends DecoratorAtaques{
    private final double probabilidad;
    private final int danioElevado;

    ProbYDanioElevado(AtaqueConcret aDecorar,double probabilidad,int danioElevado) {
        super(aDecorar);
        this.probabilidad = probabilidad * 0.01;
        this.danioElevado = danioElevado;
    }
    int especialEfecto(Chinpokomon atacado){
        int resultado;
        if( Math.random() >= this.probabilidad){
            resultado = this.getDanio() * this.danioElevado;
        }else{
            resultado = this.getDanio();
        }
        return resultado;
    }

    public boolean tieneEfecto() { return true;}
}

File: python/ProbYDanioElevado.py
import random
from DecoratorAtaques import DecoratorAtaques

class ProbYDanioElevado(DecoratorAtaques):

    def __init__(self, aDecorar, probabilidad, danioElevado) :
        super().__init__(aDecorar)
        self.probabilidad = probabilidad * 0.01
        self.danioElevado = danioElevado
    
    def especial_efecto(self, atacado):
        resultado = 0
        if random.random() >= self.probabilidad:
            resultado = self.get_danio() * self.danioElevado
        else:
            resultado = self.get_danio()        
        return resultado    

    def tiene_efecto(self):
        return True